amm-executor/test/Tests/Backlog/ServiceTest.hs

```diff
@@ -0,0 +1,70 @@
+{-# LANGUAGE RankNTypes #-}
+{-# LANGUAGE MultiParamTypeClasses #-}
+{-# LANGUAGE RecordWildCards #-}
+module Tests.Backlog.ServiceTest where
+
+import Test.Tasty
+import Test.Tasty.HUnit
+import Hedgehog
+import Spectrum.Executor.Backlog.Service (BacklogService (BacklogService, put, tryAcquire, checkLater), mkBacklogService)
+import Gen.LoggingGen (mkMakeLogging)
+import Tests.Backlog.PersistenceMock (mkMockStorage)
+import Control.Monad.Trans.Resource ( MonadResource )
+import Spectrum.Executor.Backlog.Config (BacklogServiceConfig (orderLifetime, suspendedPropability, orderExecTime, BacklogServiceConfig))
+import RIO.Time (secondsToDiffTime, NominalDiffTime, getCurrentTime, addUTCTime)
+import Spectrum.HigherKind (LiftK(liftK))
+import RIO
+import Gen.OrdersGen (genSwapOrder, genPool, genPubKeyHash)
+import qualified Hedgehog.Gen as Gen
+import qualified Hedgehog.Range as Range
+import Spectrum.Executor.Data.OrderState (OrderInState(PendingOrder, InProgressOrder))
+import Test.Tasty.Hedgehog as HH
+
+instance LiftK IO IO where
+  liftK = id
+
+
+checkBacklogService = testGroup "CheckBacklogService"
+  [ HH.testProperty "try_acquire_of_pending_orders_is_correct" getOrderTest
```

Comment:
"try_acquire_of_pending_orders_is_correct" is not a "proper" property. Please split tests into concrete properties you 're checking. E.g "tryAcquire_returns_pending_or_suspended_order_with_highest_weight".

Reply:
rename to "try_acquire_non_outdated_pending_orders"